src/Engine/include/OpenLoco/Engine/Types.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+namespace OpenLoco
+{
+    // To be replaced with std::to_underlying in c++23
+    template<typename TEnum>
+    constexpr auto enumValue(TEnum enumerator) noexcept
+    {
+        return static_cast<std::underlying_type_t<TEnum>>(enumerator);
+    }
+
+    using coord_t = int16_t;
+    using tile_coord_t = int16_t;
```

Comment:
that reminds me. Basically nothing uses tile_coord_t. It has some users for z which should use the SmallZ/MicroZ types and it has some users for a tile coordinate which should be using TilePos2.